### ISSUE TITLE
Add material on creating a Pull Request

### ISSUE THREAD

**Issue by jlperla** [QuantEcon/lecture-source-jl#260]
@arnavs  I would love to get this sort of material added at the end of the "testing" section.   But we can wait until after the lecture tomorrow and just give it to students later in the day.

At the end of the "testing" file, I think it is useful to put the workflow for creating a new pull request.  We don't need to go into the details of how to manage them, but working with them is necessary for contributing to open-source projects.

Here is a workflow that we could show with screenshots on the ExamplePackage
- In github desktop on your repo, make sure the "current branch" says "master"
- Use that dropdown  and give a title for the New branch (e.g. "Example-Changes")  and choose "Create Branch"
- Make a change to your files (i.e. we should make a code change in the tests)
- In Github desktop, choose the `Commit to Example-Changes` or whatever
- Now click "Publish Branch"
- Go to the webpage for the repo
- You will probably see a yellow warning about a recently pushed branch.  Click on "Compare & pull request".  Otherwise, you may need to use the "Branch" dropdown to pick your branch.
- In the Open a Pull Request, give it a name (usually the same as the branch) and click "Create Pull Request"
- Then we can the pull request page, show you can have discussions about it, and that it will show CI stuff.  Show that you can add "Reviewers".
- Finally, we could show the final step with the "Merge Pull Request", and point out to people that only people with the appropriate permissions would be able to do that.

If there is an easier workflow, we can adjust it

**Comment by jlperla** [comment]
One other thing, I think we should make it clear to people that they can switch between pull requests/master/etc.  just by using that dropdown, as long as they have thing committed.  So maybe add in a step where they change the branch back to master.

**Comment by jlperla** [comment]
@arnavs  I notice that you didn't implement a variation on the workflow I described in https://github.com/QuantEcon/lecture-source-jl/issues/260#issue-377985844

Was there a reason not to?

**Comment by jlperla** [comment]
@arnavs As I think I had suggested putting that workflow in the `testing.rst`, I will hand this PR back to you to complete. If you disagree with anything I wrote in the workflow, please tell me, but the explicit steps and screenshots are very important.  This stuff is not intuitive, and I spent time trying to get the minimal workflow there.

I will make a few minor modifications to what is in the `version_control.rst` lecture.

**Comment by jlperla** [comment]
I was looking a little more at what you have in the `Forking and Pull Requests` in the Version control section.  This section should be redone to ensure that it is workflow and step based rather than informational (as pretty much everything in these sections should be).

The workflow I showed above was an attempt to do that.  But one question I have is what the easiest way, in the GitHub Desktop based workflow, to fork and create a branch.

That is, I think my workflow assumed you were able to create your own branch on the repo.  If the above workflow is used, then I think there is a step required where, prior to the step that says something to `In github desktop on your repo, make sure the "current branch" says "master"` we first tell people to ensure they have a version of the repo they can modify.  If they have permissions, this could be directly off of the master, otherwise:
* Fork the repo by clicking on the fork button (as you have it)
* Clone the forked version to github desktop, as before
* Then can continue with some variation of the steps I provided

But we also need to make sure that the above steps work to make a PR for the (real) repo rather than a PR for your forked one.
